### ISSUE TITLE
при изменении масштаба Ctrl+ Ctrl- значения в контролах не возвращаются обратно

### ISSUE THREAD

**Issue by RandomAmbersky** [DarkHobbit/chronoline#24]
Когда я ставлю начало - окончание год и нажимаю Ctrl+ - Ctrl- то в контролах показывается начальное значение, а другое, видимо это из-за накопления каких-то погрешностей.

Я тут просто зафиксировал это, не срочный баг, и явно им надо заниматься где-то перед https://github.com/DarkHobbit/chronoline/issues/10, когда другого не осталось ;)

**Comment by DarkHobbit** [comment]
А возврат к начальному значению никто и не обещал :)
Там в обе стороны реализован простейший пропорциональный алгоритм масштабирования - увеличение на некую, заданную константой ZOOM_STEP из модуля cldefs.h, долю диапазона (сейчас это 10%). Соответственно, он не обратимый.
Можно ли его сделать обратимым? Да, попробовать можно, усложнив формулу масштабирования в одну из сторон (получится что-то похожее на сложные проценты или золотое сечение). Но это будет во-первых, усложнение кода, во-вторых, как бы не повлияло на быстродействие (сейчас всё же масштабируется довольно шустро), в третьих, после усложнения формы как раз может полезть то самое накопление погрешностей, т.е. не факт, насколько заметным будет улучшение.
Я на самом деле всё это обдумывал ещё когда вводил это самое масштабирование, года полтора назад. Даже додумался до радикально-экстремистской идеи запоминать стек произведённых масштабирований и при изменении направления масштабирования начинать его опустошать. Правда, при любом сдвиге шкалы его придётся сразу обнулить и начинать заполнять заново, ибо центр сместился - но это и не страшно, он "всё равно сместился". Зато вот тогда накопления погрешностей точно не будет. Правда и тормоза могут начаться из-за накопления-опустошения контейнера в реальном времени. Идея осталась нереализованной, поскольку после [этого коммита](https://github.com/DarkHobbit/chronoline/commit/368250dd41af289e81034a248d0921631eba8297) масштабирование стало НУ ОЧЕНЬ близким к обратимому, и я не стал углубляться дальше.
Ну если недостаточно близкое - будем думать.

**Comment by DarkHobbit** [comment]
Да, кроме хитровыворачивания формулы и горождения стека есть ещё третий вариант - ввести некую квазиабсолютную меру масштабирования (тот самый levelOfDetail по аналогии с чипами), и танцевать от неё, а не от предыдущего масштаба. Просто тут опять-таки ситауция сложнее, чем с чипами, из-за нелинейности временной шкалы. Короче - опять-таки надо проверять.
